packages/signature-verifier-api/src/client.rs

```diff
@@ -1,48 +1,61 @@
-use std::marker::PhantomData;
-
 use cosmwasm_schema::serde::de::DeserializeOwned;
-use cosmwasm_std::{
-    to_json_binary, Addr, CosmosMsg, Empty, HexBinary, QuerierWrapper, QueryRequest, Uint64,
-    WasmMsg, WasmQuery,
-};
+use cosmwasm_std::{CosmosMsg, HexBinary, Uint64};
 use error_stack::{Result, ResultExt};
 
 use crate::msg::{ExecuteMsg, QueryMsg};
 
-#[derive(Clone)]
-pub struct SignatureVerifier<'a, T = Empty> {
-    pub address: Addr,
-    pub querier: QuerierWrapper<'a>,
-    custom_msg_type: PhantomData<T>,
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("could not query the signature verifier contract with {signature:?}, {message:?}, {public_key:?}, {signer_address:?}, {session_id:?}")]
+    VerifySignature {
+        signature: HexBinary,
+        message: HexBinary,
+        public_key: HexBinary,
+        signer_address: String,
+        session_id: Uint64,
+    },
 }
 
-impl<'a, T> SignatureVerifier<'a, T> {
-    pub fn new(address: Addr, querier: QuerierWrapper<'a>) -> Self {
-        SignatureVerifier::<'a, T> {
-            address,
-            querier,
-            custom_msg_type: PhantomData,
+impl From<QueryMsg> for Error {
+    fn from(value: QueryMsg) -> Self {
+        match value {
+            QueryMsg::VerifySignature {
+                signature,
+                message,
+                public_key,
+                signer_address,
+                session_id,
+            } => Error::VerifySignature {
+                signature,
+                message,
+                public_key,
+                signer_address,
+                session_id,
+            },
         }
     }
+}
```

Comment:
could we change this into something like `Error::for(msg)`? I'm not a big fan of using `From` for msg to error transformations, because conceptually it's confusing. The error just "happens" to print out the information from the message, but it's not really an equivalent object